modules/core/src/main/java/org/apache/ignite/internal/IgniteNodeAttributes.java

```diff
@@ -26,6 +26,9 @@
     /** Node compound version. */
     public static final String ATTR_BUILD_VER = ATTR_PREFIX + ".build.ver";
 
+    /** Allows to override {@link #ATTR_BUILD_VER} by adding this attribute in the user attributes. */
```

Comment:
Well, it does not look a good idea to me, adding a new node attribute to `IgniteNodeAttributes` for testing purposes only. I think that required behavior can be achieved by specifying a custom `DiscoverySpi` as follows:
```
public class RollingUpgradeSelfTest extends GridCommonAbstractTest {
    /** {@inheritDoc} */
    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);

        ...

        if (nextVer != null) {
            final String nextVersion = nextVer;

            TcpDiscoverySpi testSpi = new TcpDiscoverySpi() {
                /** {@inheritDoc} */
                @Override public void setNodeAttributes(Map<String, Object> attrs, IgniteProductVersion ver) {
                    super.setNodeAttributes(attrs, IgniteProductVersion.fromString(nextVersion));
                }
            };

            cfg.setDiscoverySpi(testSpi);
        }

        return cfg;
    }
}
```